src/core/config/merge.js

```diff
@@ -34,6 +35,13 @@ const merge = (target, ...sources) => {
     merged.domNode = domNode
   }
 
+  Object.entries(merged).forEach(([key, value]) => {
+    if (key.includes(".")) {
+      delete merged[key]
+      set(merged, key, value)
```

Comment:
In the end, I used the mutable `set` here, as we're setting `domNode` in `merged` before anyway.

Reply:
> As an FYI, the immutable version changes the order of arguments to set(key, value, merged)

Yes I aware, all functions from `/fp/` abide to functional programming immutability concepts, are auto-curried and data is always last, e.g. `set(key)(value)(merged)`.